src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/Internal/ConditionsList.tsx

```diff
@@ -41,63 +65,109 @@ const ConditionsList: React.FC<IConditionsListProps> = (props) => {
     const classNames = getClassNames(styles, {
         theme: useTheme()
     });
-    const getOverflowMenuItems = (
-        conditionId: string
-    ): IContextualMenuItem[] => [
-        {
-            key: `${conditionId}-edit-menu-item`,
-            text: t('3dSceneBuilder.visualRuleForm.editCondition'),
-            iconProps: {
-                iconName: 'Edit'
-            },
-            onClick: () => {
-                alert('Edit clicked');
+
+    const [calloutInfo, setCalloutInfo] = useState<CalloutInfo>({
+        calloutType: CalloutInfoType.inactive,
+        isOpen: false,
+        selectedCondition: null,
+        selectedTarget: null
+    });
+
+    // Callbacks
+    const handleOpenNewConditionFlyout = useCallback(() => {
+        setCalloutInfo({
+            calloutType: CalloutInfoType.create,
```

Comment:
can't you tell whether to open it by when the calloutType === inactive? do we need the `isOpen` flag too?